controllers/common.go

```diff
@@ -144,16 +144,22 @@ func reconcileDesiredObjects(ctx context.Context, kubeClient client.Client, logg
 			"object_kind", desired.GetObjectKind(),
 		)
 		if isNamespaceScoped(desired) {
-			if setErr := ctrl.SetControllerReference(owner, desired, scheme); setErr != nil {
-				l.Error(setErr, "failed to set controller owner reference to desired")
-				errs = append(errs, setErr)
-				continue
+			switch desired.(type) {
+			case *rbacv1.Role, *rbacv1.RoleBinding:
+				l.Info("skipping setting controller reference for role or rolebinding")
```

Comment:
The problem is that those `Roles` and `Rolebindings` will be created from a `OpenTelemetryCollector` instance from another namespace.

Reply:
Ah, because we actually check in the prometheus receiver configuration which namespaces it wants service discovery from, and then create a Role in each of them? Makes sense.

From what I've skimmed of this PR, it looks good, but I'd appreciate if you summarized in the PR description what kinds of changes you're making and why. The original issue has a description of what the user-facing behaviour should be, but this is a sufficiently complicated changes that I'd also like to see one for the implementation.

Reply:
> From what I've skimmed of this PR, it looks good, but I'd appreciate if you summarized in the PR description what kinds of changes you're making and why. The original issue has a description of what the user-facing behaviour should be, but this is a sufficiently complicated changes that I'd also like to see one for the implementation.

Sorry for that. I tried to keep it simple but... I think the description should help much more now.

Thanks for your feedback! @swiatekm